pydnd/equipment/views.py

```diff
@@ -170,17 +172,81 @@ def get(self, request, name_or_id):
         return Response(queryset_dict, status=status.HTTP_200_OK)
 
 
-class ArmorList(APIView):
+class WeaponList(APIView):
 
     def get(self, request):
-        armor = Armor.objects.all()
-        data = ArmorListSerializer(armor, many=True).data
+        weapon = Weapon.objects.all()
+        data = WeaponListSerializer(weapon, many=True).data
         return Response(data)
 
 
-class WeaponList(APIView):
+#TODO Remove Post
+#List of equipment sub categories
+class ArmorList(generics.ListCreateAPIView):
 
-    def get(self, request):
-        weapon = Weapon.objects.all()
-        data = WeaponListSerializer(weapon, many=True).data
-        return Response(data)
\ No newline at end of file
+    def post(self, request, *args, **kwargs):
+
+        data = request.data
+
+        armor_category = get_attribute_by_name(data, 'armor_category', ArmorCategory)
+
+        armor = ArmorSerializer(data=data)
+        if armor.is_valid():
+            armor_object = armor.save()
+            armor_object.armor_category = armor_category
+            armor_object = armor_object.save()
+        else:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+        return Response(data)
+
+    queryset = Armor.objects.all()
+    serializer_class = ArmorListSerializer
+
+
+class ArmorGet(APIView):
+
+    def get(self, request, name_or_id):
+
+        if name_or_id.isdigit():
+            queryset = get_object_or_404(Armor, pk = int(name_or_id))
+        else:
+            queryset = get_object_or_404(Armor, name = name_or_id)
+
+        armor_category = get_object_or_404(ArmorCategory, pk =int(queryset.armor_category.id))
+
+        queryset_dict = {}
+
+        try:
+            armor_category_dict ={}
+            armor_category_dict["id"] = armor_category.id
+            armor_category_dict["name"] = armor_category.name
+
+            queryset_dict["id"]=queryset.id
```

Comment:
Would most of this creation of the `Armor` class work with `armor = model_to_dict(queryset)`